src/components/ClusterOverview/Details/Details.js

```diff
@@ -8,51 +11,77 @@ import {
   DashboardCardTitle,
 } from '../../Dashboard/DashboardCard';
 import { ClusterOverviewContextGenericConsumer } from '../ClusterOverviewContext';
+import { Loading } from '../../Loading';
 
-const DetailItem = ({ title, value }) => (
-  <React.Fragment>
-    <dt className="kubevirt-detail__item-title">{title}</dt>
-    <dd className="kubevirt-detail__item-value">{value}</dd>
-  </React.Fragment>
-);
+const DetailItem = ({ title, value, isLoading, LoadingComponent }) => {
+  const description = value ? (
+    <OverlayTrigger
+      overlay={<Tooltip id={`tooltip-for-${title}`}>{value}</Tooltip>}
+      placement="top"
+      trigger={['hover', 'focus']}
+      rootClose={false}
+    >
+      <span>{value}</span>
+    </OverlayTrigger>
+  ) : (
+    DASHES
+  );
+  return (
+    <React.Fragment>
+      <dt className="kubevirt-detail__item-title">{title}</dt>
+      <dd className="kubevirt-detail__item-value">{isLoading ? <LoadingComponent /> : description}</dd>
+    </React.Fragment>
+  );
+};
+
+DetailItem.defaultProps = {
+  value: null,
+};
 
 DetailItem.propTypes = {
   title: PropTypes.string.isRequired,
-  value: PropTypes.string.isRequired,
+  value: PropTypes.string,
+  isLoading: PropTypes.bool.isRequired,
+  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
 };
 
-const DetailsBody = ({ data }) => (
+const DetailsBody = ({ items, LoadingComponent }) => (
   <dl>
-    <DetailItem title="Name" value={data.name} />
-    <DetailItem title="Provider" value={data.provider} />
-    <DetailItem title="Openshift version" value={data.openshiftVersion} />
-    <DetailItem title="Docker version" value={data.dockerVersion} />
-    <DetailItem title="OS vendor" value={data.osVendor} />
+    {Object.keys(items).map(key => (
+      <DetailItem
+        key={key}
+        title={items[key].title}
+        value={items[key].value}
+        isLoading={items[key].isLoading}
+        LoadingComponent={LoadingComponent}
+      />
+    ))}
   </dl>
 );
 
 DetailsBody.propTypes = {
-  data: PropTypes.object.isRequired,
+  items: PropTypes.object.isRequired,
+  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
 };
 
-export const Details = ({ data, loaded }) => (
+export const Details = ({ items, LoadingComponent }) => (
   <DashboardCard>
     <DashboardCardHeader>
       <DashboardCardTitle>Details</DashboardCardTitle>
     </DashboardCardHeader>
-    <DashboardCardBody isLoading={!loaded}>
-      <DetailsBody data={data} />
+    <DashboardCardBody>
+      <DetailsBody items={items} LoadingComponent={LoadingComponent} />
     </DashboardCardBody>
   </DashboardCard>
 );
 
 Details.defaultProps = {
-  loaded: false,
+  LoadingComponent: Loading,
```

Comment:
not sure if it is not published yet, but please use the `InlineLoading` component here